python/lsst/analysis/ap/apdb.py

```diff
@@ -81,21 +145,33 @@ def load_sources_for_object(self, dia_object_id, limit=100000):
         data : `pandas.DataFrame`
             DiaSources for the specified diaObject.
         """
+
+        where_clauses = [f'"DiaSource"."diaObjectId" = {dia_object_id}']
+
+        if exclude_flagged:
+            where_clauses.append(self._make_flag_exclusion_clause(self.bad_DiaSource_flags))
+
+        where = "WHERE " + " and ".join(where_clauses) if len(where_clauses) else ""
```

Comment:
I almost think we should make the above block into a `self._make_where`, but it's not quite generic enough (object/forced sources don't use it)

Reply:
Let's save it for now, especially if we might come back and rewrite with SqlAlchemy.